orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/HttpClientUtils.java

```diff
@@ -160,9 +162,32 @@ public String httpGetAsString(String url) throws IOException {
     }
   }
 
+  private int getProxyPort() {
+    String proxyPort = System.getProperty(JVM_HTTP_PROXY_PORT);
+    int defaultProxyPort = 8080;
+
+    if (proxyPort != null) {
+      try {
+        return Integer.parseInt(proxyPort);
+      } catch (NumberFormatException e) {
+        LOGGER.warn(
+            "Invalid proxy port number: {}. Using default port: {}",
+            proxyPort,
+            defaultProxyPort);
+      }
+    }
+    return defaultProxyPort;
+  }
+
   class RetryRequestException extends RuntimeException {
     RetryRequestException(String cause) {
       super(cause);
     }
   }
+
+  class CustomRedirect {
+    public RedirectStrategy getLaxRedirectStrategy(){
+      return new LaxRedirectStrategy();
```

Comment:
I'm not familiar with the lax redirect strategy, but ideally failures should be a SpinnakerException vs. one of the other exceptions on failure to do the redirect.  This SHOULD bubble up the errors more effectively from an end user messaging perspective.

Reply:
I reverted to safer logic, so I left the default redirection strategy as it was before and applied a more relaxed one in case the retry mechanism is enabled and handled the exception separately. Added more tests for this.